pkg/pillar/cmd/volumemgr/handlework.go

```diff
@@ -173,7 +177,13 @@ func volumeWorker(ctxPtr interface{}, w worker.Work) worker.WorkResult {
 			// so remove pending create operation if exists
 			_ = ctx.pubVolumeCreatePending.Unpublish(vcp.Key())
 		}
-		volumeCreated, fileLocation, err = destroyVolume(ctx, d.status)
+		if d.status.FileLocation != "" {
+			volumeCreated = d.status.State == types.CREATED_VOLUME
+			fileLocation, err = handler.DestroyVolume()
```

Comment:
I do not think so. We had the same logic in `destroyVolume`+`destroyVdiskVolume`/`destroyContainerVolume` I just moved the logic to the place you mentioned to avoid additional steps and generalize `handler.DestroyVolume()`.

Reply:
I finally checked the logs and I don't see anything related to this.